### PR TITLE
Add Self-Service Teams Functionality

### LINKED ISSUE #27: Set up DB Seeding for Teams table
The teams table should include at least one default team in a clean install.

### LINKED ISSUE #334: Self-service groups/teams UI
Users should be able to join and leave groups from the web UI. Information about group membership would be available to officers as well as #48.

### PR DESCRIPTION
**Pull request by RossIV**
Closes #334 and Closes #27

The only thing _not_ done here is setting the initial team descriptions.

### CODE REVIEW

File: resources/views/layouts/app.blade.php [hunk 1]
```diff
@@ -16,6 +16,12 @@
 
         <div class="collapse navbar-collapse" id="navbarNavDropdown">
           <ul class="navbar-nav mr-auto">
+            <li class="nav-item {{ $request->is('/') ? 'active' : '' }}">
+              <a class="nav-link" href="/">Home</a>
```
Comment (kberzinch) [inline resources/views/layouts/app.blade.php]: This link should be titled Dashboard to match the title of the page itself.

File: resources/assets/js/components/teams/TeamMembershipButton.vue [hunk 2]
```diff
@@ -0,0 +1,89 @@
+<template>
+    <div>
+        <button
+                type="button" class="btn btn-secondary"
+                :disabled="!this.user.is_active || !this.team.self_serviceable"
+                v-if="memberOfTeam" v-on:click="changeMembership('leave')">
+            Leave
+        </button>
+        <button
```
Comment (kberzinch) [inline resources/assets/js/components/teams/TeamMembershipButton.vue]: Do not display a join/leave button if the user cannot join/leave the team.

Note that this will make the grey area at the bottom of the box a different size than the ones with buttons unless you do something to fix that. I think that's fine.

File: resources/assets/js/components/teams/TeamMembershipButton.vue [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                type="button" class="btn btn-primary"
+                :disabled="!this.user.is_active || !this.team.self_serviceable"
+                v-else v-on:click="changeMembership('join')">
+            Join
+        </button>
+        <em v-if="!this.user.is_active">&nbsp;&nbsp;<small>You must be an active member to join/leave teams.</small></em>
```
Comment (kberzinch) [inline resources/assets/js/components/teams/TeamMembershipButton.vue]: Do not display this message if the one below it also applies.

Also see my note about join/leave below.

File: resources/assets/js/components/teams/TeamMembershipButton.vue [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        <em v-if="!this.team.self_serviceable">&nbsp;&nbsp;<small>Join/leave is restricted for this team.</small></em>
```
Comment (kberzinch) [inline resources/assets/js/components/teams/TeamMembershipButton.vue]: Rephrase this to be more helpful - "Contact an administrator to leave this team".

If it's not too difficult, either say "join" or "leave" in the text. The slash is not necessary since the application knows which makes sense for this particular team.

File: resources/assets/js/components/teams/TeamEditForm.vue [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+            </ul>
+
+            <div class="tab-content">
+                <div class="tab-pane show active" id="members">
+                    <h3>Members</h3>
+                    <team-invite-modal
+                            id="teamInviteModal"
+                            :teamId="this.teamId">
+                    </team-invite-modal>
+                    <button type="button" class="btn btn-secondary btn-above-table" data-toggle="modal" data-target="#teamInviteModal">Invite</button>
```
Comment (kberzinch) [inline resources/assets/js/components/teams/TeamEditForm.vue]: In most contexts, an Invite is something you have to act on before you are part of a group. This button actually adds people to the team, and so it should be relabeled to indicate that.

File: resources/assets/js/components/teams/TeamInviteModal.vue [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+                    <h5 class="modal-title" :id="id">Invite Member</h5>
+                    <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                        <span aria-hidden="true">&times;</span>
+                    </button>
+                </div>
+                <div class="modal-body">
+                    <form id="inviteMemberForm" v-on:submit.prevent="submit">
+                        <div class="row">
+                            <div class="col-12">
+                                <em>Enter a first name to search...</em>
```
Comment (kberzinch) [inline resources/assets/js/components/teams/TeamInviteModal.vue]: Any part of the name will work.

File: resources/assets/js/components/teams/TeamMembershipButton.vue [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+        action: action,
+      };
+      axios
+        .post(this.baseUrl + this.team.id + '/members', data)
+        .then(response => {
+          if (response.status === 201 && data.action === 'join') {
+            swal({
+              type: 'success',
+              title: 'Sweet!',
+              text: 'You joined ' + this.team.name + '!',
```
Comment (kberzinch) [inline resources/assets/js/components/teams/TeamMembershipButton.vue]: I don't like the verbiage here but I don't know what I want it to be yet.

**Comment by kberzinch** [comment]
Can you also remove the founding year attribute in this PR?

File: resources/assets/js/components/teams/TeamMembershipButton.vue (hunk 4 above)
Comment (RossIV) [reply resources/assets/js/components/teams/TeamMembershipButton.vue]: I tried your proposed phrasing or something close to it and it was too long for the space - caused the footer panel to be larger than necessary. I'll fiddle with it some more.

I'll update the text accordingly, shouldn't be too terrible.

File: resources/assets/js/components/teams/TeamMembershipButton.vue (hunk 3 above)
Comment (RossIV) [reply resources/assets/js/components/teams/TeamMembershipButton.vue]: Note that the wording could be confusing if they then become active but still can't join/leave a non-SSE team.

Comment (kberzinch) [reply resources/assets/js/components/teams/TeamMembershipButton.vue]: What I meant was you should _only_ display the SSE message if it applies, so if a member becomes active they still don't get a join/leave button.

**Comment by RossIV** [comment]
All changes made as requested @kberzinch

File: resources/assets/js/components/teams/TeamMembershipButton.vue (hunk 7 above)
Comment (kberzinch) [reply resources/assets/js/components/teams/TeamMembershipButton.vue]: Can you add a note somewhere somehow that people should join the slack and mailing list for the team?

I want it to be clear that joining a group doesn't do anything useful (outside of SSO which I'm going to continue to handwave as magic.)

File: resources/assets/js/components/teams/TeamCreateForm.vue [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+
+                <h3>Team Details</h3>
+
+                <div class="form-group row">
+                    <label for="name" class="col-sm-2 col-form-label">Name<span style="color:red">*</span></label>
+                    <div class="col-sm-10 col-lg-4">
+                        <input v-model="team.name" type="text" class="form-control" :class="{ 'is-invalid': $v.team.name.$error }" id="name" @blur="$v.team.name.$touch()">
+                    </div>
+
+                    <label for="founding-year" class="col-sm-2 col-form-label">Founding Year<span style="color:red">*</span></label>
```
Comment (kberzinch) [inline resources/assets/js/components/teams/TeamCreateForm.vue]: Founding year needs to be removed from the UI

File: resources/assets/js/components/teams/TeamMembershipButton.vue [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+    <div>
+        <button v-if="memberOfTeam && this.user.is_active && this.team.self_serviceable"
+                type="button" class="btn btn-secondary" v-on:click="changeMembership('leave')">
+            Leave
+        </button>
+        <button v-else-if="!memberOfTeam && this.user.is_active && this.team.self_serviceable"
+                type="button" class="btn btn-primary" v-on:click="changeMembership('join')">
+            Join
+        </button>
+        <em v-else-if="this.user.is_active && !this.team.self_serviceable">&nbsp;&nbsp;
```
Comment (kberzinch) [inline resources/assets/js/components/teams/TeamMembershipButton.vue]: Display this message even if the user is not active.

File: resources/assets/js/components/teams/TeamMembershipButton.vue [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+            <small>Contact an admin to {{ this.actionVerb }} this team.</small>
+        </em>
+        <em v-else-if="!this.user.is_active">&nbsp;&nbsp;
```
Comment (kberzinch) [inline resources/assets/js/components/teams/TeamMembershipButton.vue]: Don't display this message if this team is not SSE.

File: resources/assets/js/components/teams/TeamMembershipButton.vue [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+              type: 'success',
+              title: 'Sweet!',
+              text: 'You joined ' + this.team.name + '!',
+              timer: 1500,
+            });
+            this.user_teams.push(response.data.team);
+          } else if (response.status === 201 && data.action === 'leave') {
+            swal({
+              type: 'success',
+              title: 'See you soon!',
```
Comment (kberzinch) [inline resources/assets/js/components/teams/TeamMembershipButton.vue]: Change this to "Goodbye!" or similar

**Comment by RossIV** [comment]
I did the thing @kberzinch

**Comment by kberzinch** [comment]
Use this with a team and channel ID for linking to Slack channels: https://api.slack.com/docs/deep-linking#open_a_channel

File: resources/views/teams/index.blade.php [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+                        <p class="card-text">{{ $team->description }}</p>
+                        @if($team->mailing_list_name || $team->slack_channel_id)
+                            <hr/>
+                            <p class="card-text">
+                                @if($team->mailing_list_name)
+                                    Mailing List: <a href="https://lists.gatech.edu/sympa/info/{{$team->mailing_list_name}}" target="_blank">{{$team->mailing_list_name}}</a>
+                                    <br/>
+                                @endif
+                                @if($team->slack_channel_id)
+                                    Slack: <a href="https://slack.com/app_redirect?team=T033JPZLT&channel={{$team->slack_channel_id}}">#{{$team->slack_channel_name}}</a>
```
Comment (kberzinch) [inline resources/views/teams/index.blade.php]: Open this link in a new tab.
